orga/app/services/url.js

```diff
@@ -81,12 +81,12 @@ export default class Url extends Service {
   }
 
   get forgottenPasswordUrl() {
-    const currentLanguage = this.intl.primaryLocale;
-    let url = `${this.pixAppUrlWithoutExtension}${this.currentDomain.getExtension()}/mot-de-passe-oublie`;
-    if (currentLanguage === 'en') {
-      url += '?lang=en';
+    let domain = 'fr';
+    if (!this.currentDomain.isFranceDomain) {
+      domain = 'org';
     }
-    return url;
+
+    return this.intl.t('pages.login-form.forgotten-password-url', { domain });
```

Comment:
Ne pourrait-on garder la fonction le calcul complet de l'url, et sans se baser sur la traduction, dans `forgottenPasswordUrl` mais en tirant parti de la langue de manière dynamique ?


```suggestion
    const currentLanguage = this.intl.primaryLocale;
    let url = `${this.pixAppUrlWithoutExtension}${this.currentDomain.getExtension()}/mot-de-passe-oublie`;
    url += `?lang=${currentLanguage}`;
    return url;
```

Reply:
On pourrait, mais c'est beaucoup d'effort pour recréer une url qui est déjà présente dans le fichier de trad non ?

Du coup on fait un getter qui calcul tout et on retire les clés de trad ?

Reply:
> Du coup on fait un getter qui calcul tout et on retire les clés de trad ?

Moi je suis pour ça oui.

Les traducteurs ne devraient pas avoir à gérer des URL. Et en plus ils peuvent facilement faire des erreurs en copiant des URL … ou pire … vouloir traduire les URL :scream:

Reply:
Je rejoins @lego-technix, on a vu dans des tickets précédents que gérer les URLs dans les fichiers de traduction n'est vraiment pas pratique:
- au niveau couplage et complexité: les fichiers de trad son conçu pour gérer le texte et pas la logique applicative des URLs, quand elles changent, on doit éditer tous les fichiers de traduction (et plus on aura de langue, plus se sera difficile)
- le manque de flexibilité: parfois les URLs nécessitent d'être dynamiques (query params...)

Le ticket est l'occasion de corriger ce problème pour cette URL ;)

Reply:
Ça roule 🙂